protocol/x/vault/keeper/vault_test.go

```diff
@@ -0,0 +1,135 @@
+package keeper_test
```

Comment:
non-blocking nit: Both tests here can be broken up into separate test cases w/ descriptions which would be more helpful for someone running the test if anything fails.